components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/RetrieveRuntimeArtifactsApiServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.wso2.carbon.apimgt.internal.service.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.cxf.jaxrs.ext.MessageContext;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.ExceptionCodes;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.dto.RuntimeArtifactDto;
+import org.wso2.carbon.apimgt.impl.gatewayartifactsynchronizer.RuntimeArtifactGeneratorUtil;
+import org.wso2.carbon.apimgt.internal.service.RetrieveRuntimeArtifactsApiService;
+import org.wso2.carbon.apimgt.internal.service.dto.SynapseArtifactListDTO;
+import org.wso2.carbon.apimgt.internal.service.utils.SubscriptionValidationDataUtil;
+import org.wso2.carbon.apimgt.rest.api.common.RestApiConstants;
+import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.StreamingOutput;
+import java.io.File;
+import java.nio.file.Files;
+import java.util.List;
+
+/**
+ * Runtime Artifact Service implementation.
+ */
+public class RetrieveRuntimeArtifactsApiServiceImpl implements RetrieveRuntimeArtifactsApiService {
+
+    @Override
+    public Response retrieveRuntimeArtifactsGet(String xWSO2Tenant, String type, String dataPlaneId,
+                                                MessageContext messageContext) throws APIManagementException {
+        RuntimeArtifactDto runtimeArtifactDto;
+        xWSO2Tenant = SubscriptionValidationDataUtil.validateTenantDomain(xWSO2Tenant, messageContext);
```

Comment:
shall we remove the `xWSO2Tenant` and related logic? This will not be used in Choreo.

Reply:
I have removed the ability to send the xWSO2Tenant as a header to the resource. But we cannot completely remove this as we rely on this to check whether the request is coming from the super tenant user when we use the `ALL` in the organization parameter.

Reply:
Removed xWSO2Tenant validation as per the offline discussion.